### PR TITLE
Feature/tfn 406 flat fare user journey

### PR DESCRIPTION
**Pull request by dannydaviesIW**
# Description

-   This PR introduces a new user journey for entering a Flat Fare Product.
-   This PR also includes some housekeeping to improve consistency between web pages

# Testing instructions

-   Pull down this branch and run all resources locally using the make command in fdbt-dev
-   Run through the new Flat Fare Product user journey (see the sitemap in https://drive.google.com/drive/u/0/folders/0AFrrMIylp35DUk9PVA for more detail) and validate that the 'fareType', 'serviceList' and 'productDetails' pages route correctly.
-   Run through the existing Period Ticket user journeys to validate that the existing pages route as they did in the past.
- Run `awslocal s3api list-objects --bucket fdbt-matching-data-dev` to find the file which has been dumped in s3local by the different user journeys. Then run `awslocal s3api get-object --bucket fdbt-matching-data-dev --key <FILENAME IN S3LOCAL> <FILEPATH TO SAVE TO>`

# Type of change

-   [X] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

# Checklist:

-   [X] Able to run pr locally
-   [X] Lighthouse performed
-   [X] Followed acceptance criteria
-   [X] My code follows the style guidelines of this project
-   [X] I have performed a self-review of my own code
-   [X] I have made corresponding changes to the documentation if applicable
-   [X] I have added tests that prove my fix is effective or that my feature works

Screenshot 2020-05-15 at 15 21 06
Screenshot 2020-05-15 at 15 21 41
Screenshot 2020-05-15 at 15 23 27
Screenshot 2020-05-15 at 15 23 51

### CODE REVIEW

**Comment by tejpowar** [comment]
Linting fails in circle CI

File: src/pages/thankyou.tsx [hunk 1]
```diff
@@ -4,19 +4,21 @@ import Layout from '../layout/Layout';
 import { FEEDBACK_LINK } from '../constants';
 import { getUuidFromCookies, deleteAllCookiesOnServerSide } from '../utils';
 
-const title = 'Thank You - Fares data build tool';
-const description = 'Thank you page for the Fares Data Build Tool';
+const title = 'Thankyou - Fares Data Build Tool';
```
Comment (ljones92) [inline src/pages/thankyou.tsx:7]: I would keep thank you as two separate words

File: tests/pages/api/multipleProductValidity.test.ts [hunk 2]
```diff
@@ -40,7 +89,26 @@ describe('multipleProductValidity API', () => {
...
+    it('generates and dumps multiple period product JSON in S3 when the user selects a list of services', async () => {
+        const { req, res } = getMockRequestAndResponse(
+            { fareType: 'period', numberOfProducts: '3', fareZoneName: null },
+            { 'validity-row0': '24hr', 'validity-row1': '24hr', 'validity-row2': 'endOfCalendarDay' },
+            '',
+            writeHeadMock,
+        );
+        await multipleProductValidity(req, res);
+        const actualMultipleProductData = JSON.parse((putStringInS3Spy as jest.Mock).mock.calls[0][2]);
+        console.log(actualMultipleProductData);
```
Comment (ljones92) [inline tests/pages/api/multipleProductValidity.test.ts:104]: Remove this

File: src/pages/api/periodValidity.ts [hunk 3]
```diff
@@ -127,6 +135,7 @@ export default async (req: NextApiRequest, res: NextApiResponse): Promise<void>
         }
     } catch (error) {
         const message = 'There was a problem selecting the period validity:';
+        console.log(error);
```
Comment (tejpowar) [inline src/pages/api/periodValidity.ts:138]: remove console

File: src/pages/api/productDetails.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        if (fareType === 'period') {
+            const validInputs = JSON.stringify(productDetails);
+            setCookieOnResponseObject(getDomain(req), PRODUCT_DETAILS_COOKIE, validInputs, req, res);
+            redirectTo(res, '/chooseValidity');
+        } else if (fareType === 'flatFare') {
+            const operatorCookie = unescapeAndDecodeCookie(cookies, OPERATOR_COOKIE);
+            const serviceListCookie = unescapeAndDecodeCookie(cookies, SERVICE_LIST_COOKIE);
+
+            const { operator, uuid, nocCode } = JSON.parse(operatorCookie);
+            const { selectedServices } = JSON.parse(serviceListCookie);
```
Comment (tejpowar) [inline src/pages/api/productDetails.ts:78]: Add check to see if selectServices is valid

File: tests/pages/productDetails.test.tsx [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    describe('productDetails', () => {
+        it('should render correctly', () => {
+            const tree = shallow(
+                <ProductDetails product={mockproductDetails} operator="bus company" hintText="Test Zone" />,
+            );
+            expect(tree).toMatchSnapshot();
+        });
+    });
+
+    describe('productDetailsGetServerSideProps', () => {
```
Comment (ljones92) [inline tests/pages/productDetails.test.tsx:24]: I would keep this as two separate words 'productDetails getServerSideProps' or nest a 'getServerSideProps' describe within the 'productDetails' describe

File: tests/pages/serviceList.test.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        });
+
+        it('should render an error when the error flag is set to true', () => {
+            // eslint-disable-next-line react/jsx-props-no-spreading
+            const tree = shallow(<ServiceList {...serviceInfoWithError} />);
+            expect(tree).toMatchSnapshot();
+        });
+    });
+
+    describe('serviceListGetServerSideProps', () => {
```
Comment (ljones92) [inline tests/pages/serviceList.test.tsx:51]: Same here

File: src/pages/api/service/validator.ts [hunk 7]
```diff
@@ -46,28 +39,21 @@ export const isCookiesUUIDMatch = (req: NextApiRequest, res: NextApiResponse): b
 
 export const isPeriodCookiesUUIDMatch = (req: NextApiRequest, res: NextApiResponse): boolean => {
```
Comment (RobCatton) [inline src/pages/api/service/validator.ts:40]: Is this named properly now?

Comment (dannydaviesIW) [reply src/pages/api/service/validator.ts:40]: Removed this now

File: tests/pages/productDetails.test.tsx (hunk 5 above)
Comment (dannydaviesIW) [reply tests/pages/productDetails.test.tsx]: Separated this and nested within the productDetails describe

File: tests/pages/serviceList.test.tsx (hunk 6 above)
Comment (dannydaviesIW) [reply tests/pages/serviceList.test.tsx]: Same here

File: src/pages/thankyou.tsx (hunk 1 above)
Comment (dannydaviesIW) [reply src/pages/thankyou.tsx]: Reverted back to having these as separate words

**Pull request merged**